fxola/Banka
# #165127799 Write tests for user sign up and sign in endpoints

#### What does this PR do?
- Write tests for the POST  ```api/v1/auth/signup``` and ```api/v1/auth/signin``` endpoints


#### Description of Task to be completed?
- install necessary dependencies needed to write tests
- write test cases for user sign up endpoint
- write test cases for user sign in endpoint
- refactor user and transaction middleware validations to remove 'Cannot
set headers after they are sent to the client' errors from the console
- delivers[#165127799]

#### How should this be manually tested?
- Clone the repository using ```git clone https://github.com/fxola/Banka.git .``` in your terminal
- run ```npm install``` on your terminal
- run ```npm run start``` to start up the server
- run ```npm test```  to see test results

#### Any background context you want to provide?
- Create and environment variable with the format ```SECRET = yoursecret``` in the root of the server directory


#### What are the relevant pivotal tracker stories?
- PT Story link - [#165127799] (https://www.pivotaltracker.com/n/projects/2321791/stories/165127799)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A
